### PR TITLE
Remote page component

### PR DESCRIPTION
**Pull request by sharanyavinod**
- Add a new custom page component to load remote SPA scripts and render app within editor
- Remote spa origin to be provided via page properties (_remoteSPAUrl_)

### CODE REVIEW

**Review by grubyak — APPROVED** [review]

**Review by CezCz — APPROVED** [review]

File: core/src/main/java/com/adobe/aem/spa/project/core/internal/impl/PageImpl.java [hunk 1]
```diff
@@ -247,6 +255,12 @@ public String getExportedType() {
         return delegate.getExportedType();
     }
 
+    // Delegated to Page v1
```
Comment (habansal) [inline core/src/main/java/com/adobe/aem/spa/project/core/internal/impl/PageImpl.java:258]: are we changing text from v2 to v1 here ?

**Review by habansal — APPROVED** [review]

File: core/src/main/java/com/adobe/aem/spa/project/core/internal/impl/PageImpl.java [hunk 2]
```diff
@@ -67,12 +68,16 @@
...
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = { Page.class, ContainerExporter.class },
+    resourceType = { PageImpl.RESOURCE_TYPE, PageImpl.RESOURCE_TYPE_REMOTE  }
+    )
 @Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME, extensions = ExporterConstants.SLING_MODEL_EXTENSION)
 public class PageImpl implements Page {
 
     static final String RESOURCE_TYPE = "spa-project-core/components/page";
+    static final String RESOURCE_TYPE_REMOTE = "spa-project-core/components/remotepage";
```
Comment (pfauchere) [inline core/src/main/java/com/adobe/aem/spa/project/core/internal/impl/PageImpl.java:80]: Hmm, you are reusing the same Sling Model for both resource types ... interesting and practical. But shouldn't you extend the Java class instead?

File: core/src/main/java/com/adobe/aem/spa/project/core/models/Page.java [hunk 3]
```diff
@@ -105,4 +105,11 @@ default String getExportedPath() {
     default Map<String, ? extends HierarchyNodeExporter> getExportedChildren() {
         throw new UnsupportedOperationException();
     }
+
+    /**
+     * @return the remote SPA page's URL, if one was set, or {@code null}
+     */
+    default String getRemoteSPAUrl() {
```
Comment (pfauchere) [inline core/src/main/java/com/adobe/aem/spa/project/core/models/Page.java:112]: The same way, shouldn't we extend the interface?

Comment (pfauchere) [reply core/src/main/java/com/adobe/aem/spa/project/core/models/Page.java:112]: Not all the HierarchyPages should have this field, right?

File: ui.apps/src/main/content/jcr_root/apps/spa-project-core/components/remotepage/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Rendering remote SPA in AEM
+By default, the remotepage component creates an element -
+
+`<div id="root"></div>`
+
+ which will behave as the root DOM node in which the react application is rendered.
+If the remote application to be edited has a different root DOM element/id, please override the [body.html](./body.html) in the extended component.
+
+## Restrictions
```
Comment (pfauchere) [inline ui.apps/src/main/content/jcr_root/apps/spa-project-core/components/remotepage/README.md:35]: I would call it: _Limitations_ or _Constraints_. WDYT?

File: ui.apps/src/main/content/jcr_root/apps/spa-project-core/components/remotepage/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+For eg:  for a remote react application at `https://test.com` with the generated asset manifest at `https://test.com/asset-manifest.json` the URL to be provided would be `https://test.com`.
+
+## Rendering remote SPA in AEM
+By default, the remotepage component creates an element -
+
+`<div id="root"></div>`
+
+ which will behave as the root DOM node in which the react application is rendered.
+If the remote application to be edited has a different root DOM element/id, please override the [body.html](./body.html) in the extended component.
+
```
Comment (pfauchere) [inline ui.apps/src/main/content/jcr_root/apps/spa-project-core/components/remotepage/README.md:34]: What about a _Requirements_ section?

That would elaborate on the need to have URLs be absolute?

**Review by pfauchere — APPROVED** [review]

**Pull request merged**